src/rngs/adapter/reseeding.rs

```diff
@@ -136,8 +140,9 @@ where R: BlockRngCore + SeedableRng,
     type Results = <R as BlockRngCore>::Results;
 
     fn generate(&mut self, results: &mut Self::Results) {
-        if self.bytes_until_reseed <= 0 {
-            // We get better performance by not calling only `auto_reseed` here
+        if self.bytes_until_reseed <= 0 ||
+           self.fork_counter < get_fork_counter() {
```

Comment:
I wonder if we'd get better performance by using binary OR (`|`) to allow calculation of both conditions simultaneously? Normally both would be false.